docs/core/config/auth/databases/ldap.md

```diff
@@ -268,77 +283,97 @@ https://technet.microsoft.com/en-us/library/cc978012.aspx.
-`/etc/dovecot/dovecot-ldap.conf.ext.` Its most important settings are:
-
-Configure how the LDAP server is reached.
+- Configure how the LDAP server is reached.
 (Active directory allows binding with username@domain):
 
+::: code-group
+```[dovecot.conf]
+  ldap_uris = ldap://ldap.example.com
+  ldap_auth_dn = cn=admin,dc=example,dc=com
+  ldap_auth_dn_password = secret
+  ldap_base = dc=example,dc=com
 ```
-hosts = ldap.example.com
-dn = cn=admin,dc=example,dc=com
-dnpass = secret
-base = dc=example,dc=com
-```
+:::
 
-Use LDAP authentication binding for verifying users' passwords:
+- Use LDAP authentication binding for verifying users' passwords:
 
+::: code-group
+```[dovecot.conf]
+  passdb_ldap_bind_userdn = %u
+  passdb_ldap_bind = yes
 ```
-auth_bind_userdn = %u
-auth_bind = yes
-```
+:::
 
-Use auth worker processes to perform LDAP lookups in order to use multiple
+- Use auth worker processes to perform LDAP lookups in order to use multiple
 concurrent LDAP connections. Otherwise only a single LDAP connection is used.
 
+::: code-group
+```[dovecot.conf]
+(passdb|userdb) ... {
+  use_worker = yes
+}
 ```
-blocking = yes
-```
+:::
 
-Normalize the username to exactly the `mailRoutingAddress` field's value
-regardless of how the `pass_filter` found the user:
+- Normalize the username to exactly the `mailRoutingAddress` field's value
+regardless of how the `ldap_filter` found the user:
 
+::: code-group
+```[dovecot.conf]
+  fields {
+      user     = %{ldap:mailRoutingAddress}
+      password = %{ldap:userPassword}
+      proxy    = y
+      proxy_timeout = 10
+  }
 ```
-pass_attrs =                          \
-    =proxy=y,                         \
-    =proxy_timeout=10,                \
-    =user=%{ldap:mailRoutingAddress}, \
-    =password=%{ldap:userPassword}
-```
+:::
 
-Returns userdb fields when prefetch userdb wasn't used (LMTP & doveadm).
+- Returns userdb fields when prefetch userdb wasn't used (LMTP & doveadm).
 The username is again normalized in case `user_filter` found it via some
 other means:
 
+::: code-group
+```[dovecot.conf]
+  fields {
+    user = %{ldap:mailRoutingAddress}
+    quota_rule = *:storage=%{ldap:messageQuotaHard}
+  }
 ```
-user_attrs =                          \
-    =user=%{ldap:mailRoutingAddress}, \
-    =quota_rule=*:storage=%{ldap:messageQuotaHard}
-```
+:::
 
-How to find the user for passdb lookup:
+- How to find the user for passdb lookup (this can be set specifically to
+distinct values inside each passdb/userdb stanza):
 
+::: code-group
+```[dovecot.conf]
+ldap_filter = (mailRoutingAddress=%u)
 ```
-pass_filter = (mailRoutingAddress=%u)
-user_filter = (mailRoutingAddress=%u)
-```
+:::
 
-How to iterate through all the valid usernames:
+- How to iterate through all the valid usernames:
 
+::: code-group
+```[dovecot.conf]
+  ldap_filter = (mailRoutingAddress=%u)
+  ldap_iterate_filter = (objectClass= messageStoreRecipient)
```

Comment:
weird space in here, not sure if it's even allowed (I see it was in the original text already)